cmd/sonarExecuteScan_test.go

```diff
@@ -0,0 +1,303 @@
+package cmd
+
+import (
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	FileUtils "github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type mockDownloader struct {
+	shouldFail    bool
+	requestedURL  []string
+	requestedFile []string
+}
```

Comment:
Not important for this PR, but I have pretty much the exact same thing in the nexusUpload tests. Something for pkg/mock eventually.

Reply:
You're right, added a TODO about it.